packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/modeledDataStreamTable/modeledDataStreamTable.tsx

```diff
@@ -130,6 +136,7 @@ export function ModeledDataStreamTable({
       stripedRows={preferences.stripedRows}
       wrapLines={preferences.wrapLines}
       stickyColumns={preferences.stickyColumns}
+      isItemDisabled={(item) => isLineBarWidgetSelected && hasNotAllowedDataType(item.dataType)}
```

Comment:
this works fine for line and bar... can we make these functions more generic given we will have validation for all widget types very soon. if you can create a pattern, we can just add new validations on top.. i.e try to create a generic function which take widget type and other inputs and returns a simple boolean... all the logic like in line 50-51 and 116-118 will live inside this "`isValidProperty`" function..

Reply:
changes updated.